include/spirv-tools/libspirv.h

```diff
@@ -474,6 +480,19 @@ spv_result_t spvValidateBinary(const spv_const_context context,
                                const uint32_t* words, const size_t num_words,
                                spv_diagnostic* diagnostic);
 
+spv_result_t spvLink(const spv_const_context context,
```

Comment:
* Regarding the versions, I was assuming that a SPIR-V 1.0 binary could be marked as 1.1, except if instructions are deprecated, or their format changed but the latter should never happen (I guess). Therefore, I was making the linked binary use the maximum of all encountered versions among the binaries given as input.
* I was assuming that the input binaries had already been validated, to avoid having validation checks in the linker code. However, if I use `spvBinaryParse` or the IR from the optimiser, that might not be a problem.
  I guess validation would fail for a Vulkan module as Linkage requires Addresses, which itself requires a non-logical memory model, which is not supported by the Vulkan model?